delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/deletion/ProjectDeletionCoordinator.scala

```diff
@@ -0,0 +1,140 @@
+            )
+            delete(project).retryingOnSomeErrors(retryStrategy.retryWhen, retryStrategy.policy, retryStrategy.onError)
+          case _                                    => Task.unit
+        }
+      }
+
+    private[deletion] def delete(project: ProjectState) =
+      for {
+        _         <- Task.delay(logger.warn(s"Starting deletion of project ${project.project}"))
+        now       <- IOUtils.instant
+        // Running preliminary tasks before deletion like deprecating and stopping views,
+        // removing acl related to the project, etc...
+        initReport = ProjectDeletionReport(project.project, project.updatedAt, now, project.updatedBy)
+        report    <- deletionTasks
+                       .foldLeftM(initReport) { case (report, task) =>
+                         task(project.project).map(report ++ _)
+                       }
+        // Waiting for events issued by deletion tasks to be taken into account
+        _         <- Task.sleep(deletionConfig.propagationDelay)
+        // Delete the events and states and save the deletion report
+        _         <- deletionStore.deleteAndSaveReport(report)
+        _         <- Task.delay(logger.info(s"Project ${project.project} has been successfully deleted."))
+      } yield ()
+
+    /**
+      * Save the deletion report for the given project
+      */
+    private[deletion] def list(project: ProjectRef): Task[List[ProjectDeletionReport]] =
+      deletionStore.list(project)
+  }
+
+  /**
+    * Build the project deletion stream according to the configuration
+    */
+  def apply(
+      projects: Projects,
+      deletionTasks: Set[ProjectDeletionTask],
+      deletionConfig: ProjectsConfig.DeletionConfig,
+      serviceAccount: ServiceAccount,
+      xas: Transactors
+  )(implicit clock: Clock[UIO]): ProjectDeletionCoordinator =
+    if (deletionConfig.enabled) {
+      new Active(
+        projects.states,
+        deletionTasks.toList,
+        deletionConfig,
+        serviceAccount,
+        new ProjectDeletionStore(xas)
+      )
+    } else
+      Noop
+
+  /**
+    * Build and run the project deletion stream in the supervisor
+    */
+  // $COVERAGE-OFF$
+  def apply(
+      projects: Projects,
+      deletionTasks: Set[ProjectDeletionTask],
+      deletionConfig: ProjectsConfig.DeletionConfig,
+      serviceAccount: ServiceAccount,
+      supervisor: Supervisor,
+      xas: Transactors
+  )(implicit clock: Clock[UIO]): Task[ProjectDeletionCoordinator] = {
+    val stream = apply(projects, deletionTasks, deletionConfig, serviceAccount, xas)
+    stream match {
+      case Noop           => Task.delay(logger.info("Projection deletion is disabled.")).as(Noop)
+      case active: Active =>
+        val metadata: ProjectionMetadata = ProjectionMetadata("system", "project-deletion", None, None)
+        supervisor
+          .run(
+            CompiledProjection.fromStream(
+              metadata,
+              ExecutionStrategy.PersistentSingleNode,
```

Comment:
Question: could this be Transient?

Reply:
This way, there is a little performance boost as we don't have to start from the beginning of the log.
It can also be interesting for monitoring/debugging